amads/melody/contour/polynomial_contour.py

```diff
@@ -0,0 +1,246 @@
+import numpy as np
+
+from amads.core.basics import Note, Score
+
+
+class PolynomialContour:
+    """A class for computing polynomial contour, as described in the FANTASTIC toolbox [1].
+    This approach is discussed in detail in Müllensiefen and Wiggins (2009) [2].
+
+    Attributes
+    ----------
+    score : Score
+        The score object containing the melody to analyze.
+    _coefficients : list[float]
+        The coefficients of the polynomial contour. This is a private attribute and should
+        not be accessed directly. Instead, use the `coefficients` property.
+
+    References
+        ----------
+        [1] Müllensiefen, D. (2009). Fantastic: Feature ANalysis Technology Accessing
+        STatistics (In a Corpus): Technical Report v1.5
+        [2] Müllensiefen, D., & Wiggins, G. A. (2009). Polynomial functions as a representation of
+        melodic phrase contour
+
+    Examples
+    --------
+    >>> the_lick = Score.from_melody([62, 64, 65, 67, 64, 60, 62], [1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0])
+    >>> pc = PolynomialContour(the_lick)
+    >>> pc.coefficients  # this value is confirmed by the FANTASTIC toolbox
+    [-1.5014826, -0.2661533, 0.1220570]
+
+    >>> twinkle_twinkle = Score.from_melody([60, 60, 67, 67, 69, 69, 67, 65, 65, 64, 64, 62, 62, 60],
+    ... [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 2.0])
+    >>> pc2 = PolynomialContour(twinkle_twinkle)
+    >>> pc2.coefficients # there is a small mismatch with the FANTASTIC toolbox - TODO: investigate
+    [-0.9535562, 0.2120971, 0.0000000]
+    """
+
+    def __init__(self, score: Score):
+        self.score = score
+        self._coefficients = None
+
+    @property
+    def coefficients(self) -> list[float]:
+        """Get the polynomial contour coefficients. Only the three non-constant coefficients
```

Comment:
Why would you not compute the coefficients immediately in `__init__`? I'm struggling to think of a situation where you'd not want to do that.